lib/node/index.js

```diff
@@ -409,14 +409,12 @@ Request.prototype.pipe = function(stream, options){
 Request.prototype._pipeContinue = function(stream, options){
   this.req.once('response', res => {
     // redirect
-    const redirect = isRedirect(res.statusCode);
-    if (redirect && this._redirects++ != this._maxRedirects) {
-      if (!res.headers.location) {
-        this.callback(new Error('No location header for redirect'), res);
-        stream.end();
```

Comment:
On second thought, superagent should not be responsible for ending the client stream if there is an error. Instead the client should listen for the `error` event and then handle their stream on their own.

Reply:
please advise...